specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+
+- by the deposit transactions included in the L1 block, if the payloads are for the first block of
+the current sequencing epoch,
+- by an empty list otherwise.
+
+As before, a failure to then process the deposit-only attributes is a critical error.
+
+If an invalid payload is replaced by a deposit-only payload, for consistency reasons, the remaining
+span batch, if applicable, and channel it originated from are dropped as well.
+
+## Activation
+
+The new batch rules activate when the _L1 inclusion block timestamp_ is greater or equal to the
+Holocene activation timestamp. Note that this is in contrast to how span batches activated in
+[Delta](../delta/overview.md), namely via the span batch L1 origin timestamp.
+
+When the L1 traversal stage of the derivation pipeline moves its origin to the L1 block whose
+timestamp matches the Holocene activation timestamp, the derivation pipeline's state is mostly reset
+by **discarding**
+
+- all frames in the frame queue,
+- channels in the channel bank, and
+- all batches in the batch queue.
+
+The three stages are then replaces by the new Holocene frame queue, channel bank and batch queue
+(and, depending on the implementation, the optional span batch stage is added).
+
+Note that batcher implementations must be aware of this activation behavior, so any frames of a
+partially submitted channel that were included pre-Holocene must be sent again. This is a very
+unlikely scenario since production batchers are usually configured to submit a channel in a single
+transaction.
+
+## Sync Start
+
+Thanks to the new strict frame and batch ordering rules, the sync start algorithm can be simplified in the
+average case. The rules guarantee that
+
+- an incoming first frame for a new channel leads to discarding previous incomplete frames for a
+non-closed previous channel in the frame queue and channel bank, and
+- when the derivation pipeline L1 origin progresses, the batch queue is empty.
+
+So the sync start algorithm can optimistically select the last L2 unsafe, safe and finalized heads
+from the engine and if the L2 safe head's L1 origin is _plausible_ (see the
+[original sync start description](../derivation.md#finding-the-sync-starting-point) for details),
+start deriving from this L1 origin.
+
+- If the first frame we find is a _first frame_ for a channel that includes the safe head (TODO: or
+even just the following L2 block with the current safe head as parent?), we can
+safely continue derivation from this channel because no previous derivation pipeline state could
+have influenced the L2 safe head.
+- If the first frame we find is a non-first frame, then we need to go back a full channel
+timeout window to see if we find the start of that channel.
+  - If we find the starting frame, we can continue derivation from it.
+  - If we don't find the starting frame, we need to go back a full sequencing window. (TODO: verify)
```

Comment:
The batch for an L2 block is only valid to be included on L1 _after_ the L2 block's L1 origin. So if we don't find a starting frame, the channel we did find a frame from must be timed out and would be discarded.  The block we're looking for can't be in any channel that timed out before its L1 origin so we shouldn't need to search any further back.

Even pre-holocene we don't search back a full sequencing window if we hit the finalized L2 block first - worst case is one ChannelTimeout prior to the L1 origin of the L2 finalized head.

Reply:
That makes sense, fixed and added with explanation, thanks!